### ISSUE TITLE
Implement Model Performance Metrics for Binary Classification

### ISSUE THREAD

**Issue by cce-dev-account** [cce-dev-account/segmentation_toolkit#1]
**Priority:** 🔴 P1 Critical  
**Effort:** 2-3 days  
**Labels:** enhancement, irb-methodology, metrics, priority-high

## Description
The toolkit currently lacks key discriminatory power metrics required for IRB model validation. While `validators.py:23` references 'ks' and 'gini' as valid tests, they are not implemented in `run_validation_tests()`.

**Required Metrics (Binary Classification):**
- Gini coefficient
- KS statistic (Kolmogorov-Smirnov)
- AUC (Area Under ROC Curve)
- Accuracy Ratio (AR)
- Brier Score

**Regulatory Context:**
Basel II/III IRB requirements mandate quantitative validation of discriminatory power. Supervisory authorities expect Gini > 0.30 and KS > 0.20 for acceptable models.

## Implementation Plan

**File:** `irb_segmentation/validators/performance_metrics.py` (new)

```python
"""
Model Performance Metrics for Binary Classification

Implements discriminatory power metrics required for IRB validation.
Designed for binary PD models with forward compatibility for regression.
"""

import numpy as np
from typing import Tuple, Dict
from scipy import stats
from sklearn.metrics import roc_auc_score, roc_curve

def gini_coefficient(y_true: np.ndarray, y_pred_proba: np.ndarray) -> float:
    """
    Calculate Gini coefficient (2*AUC - 1).
    
    Range: [-1, 1], where 1 = perfect discrimination
    Basel threshold: > 0.30 for acceptable model
    
    Note: Currently supports binary classification only.
    Future: Extend for continuous targets (LGD/EAD).
    """
    auc = roc_auc_score(y_true, y_pred_proba)
    return 2 * auc - 1

def ks_statistic(y_true: np.ndarray, y_pred_proba: np.ndarray) -> Tuple[float, float]:
    """
    Kolmogorov-Smirnov statistic.
    
    Returns:
        (ks_stat, ks_threshold) - max separation between cumulative distributions
    
    Basel threshold: > 0.20 for acceptable model
    """
    # Separate scores by class
    scores_pos = y_pred_proba[y_true == 1]
    scores_neg = y_pred_proba[y_true == 0]
    
    # KS test
    ks_stat, _ = stats.ks_2samp(scores_pos, scores_neg)
    
    # Find threshold at max separation
    fpr, tpr, thresholds = roc_curve(y_true, y_pred_proba)
    ks_idx = np.argmax(tpr - fpr)
    ks_threshold = thresholds[ks_idx]
    
    return ks_stat, ks_threshold

def brier_score(y_true: np.ndarray, y_pred_proba: np.ndarray) -> float:
    """
    Brier score (mean squared error of probabilities).
    
    Range: [0, 1], where 0 = perfect calibration
    """
    return np.mean((y_true - y_pred_proba) ** 2)

def accuracy_ratio(y_true: np.ndarray, y_pred_proba: np.ndarray) -> float:
    """
    Accuracy Ratio = Gini / Gini_perfect.
    
    Used in some regulatory frameworks as alternative to Gini.
    """
    gini = gini_coefficient(y_true, y_pred_proba)
    # Perfect Gini when all positives ranked first
    n_pos = y_true.sum()
    n_total = len(y_true)
    gini_perfect = 1 - (n_pos / n_total)
    
    return gini / gini_perfect if gini_perfect > 0 else 0.0
```

**Integration Point:** `irb_segmentation/validators/validators.py:95`

Update `run_validation_tests()` to call these metrics when 'ks' or 'gini' specified in `validation_tests`.

## Acceptance Criteria
- [ ] Implement all 5 metrics with unit tests (90%+ coverage)
- [ ] Integrate into `run_validation_tests()` validator
- [ ] Add regulatory thresholds as configurable parameters
- [ ] Include metrics in segmentation reports (JSON/Excel)
- [ ] Document metric interpretation in `docs/metrics.md`
- [ ] ⚠️ Note in documentation: "Currently supports binary classification. Regression metrics for LGD/EAD planned (see Issue #6)"

## References
- Basel II IRB Framework (paragraphs 417-418)
- EBA Guidelines on PD estimation (EBA/GL/2017/16)
- sklearn metrics: https://scikit-learn.org/stable/modules/model_evaluation.html
